src/infill.cpp

```diff
@@ -4,9 +4,20 @@
 #include "utils/polygonUtils.h"
 #include "utils/logoutput.h"
 
-namespace cura {
+/*!
+ * Function which returns the scanline_idx for a given x coordinate
+ *
+ * For negative \p x this is different from simple division.
+ *
+ * \warning \p line_width is assumed to be positive
+ *
+ * \param x the point to get the scansegment index for
+ * \param line_width the width of the scan segments
+ */
+static inline int computeScanSegmentIdx(int x, int line_width);
```

Comment:
I think putting all function declarations in the header provides an overview of each function with its documentation, which is also valuable.

It may be a matter of taste, but in the interest of code consistency we've specified in our code conventions that function declarations and their documentation go in the header file: https://github.com/Ultimaker/Meta/blob/master/code_conventions.md#documentation